tests/unit/cli/test_download_args_parser.py

```diff
@@ -66,6 +66,11 @@ class TestDownloadArgsParser:
                 "dl --youtube.playlist_url https://youtube.com/playlist?list=123abc",
                 {"youtube": {"playlist_url": "https://youtube.com/playlist?list=123abc"}},
             ),
+            (
+                None,
+                "dl --parameter.using.list._2 'v2' --parameter.using.list._1 'v1'",
+                {"parameter": {"using": {"list" : ['v1', 'v2']}}},
```

Comment:
Lets add some more tests, like
- only one list argument
- argument indices in order (_1, _2)
- duplicate indices (_2, _1, _2)